File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        from collections import defaultdict

        hashMap = defaultdict(int)
        arr = [[] for _ in range(len(nums) + 1)] 

        for _, num in enumerate(nums):
            hashMap[num] += 1

        for key, freq in hashMap.items():
            arr[freq].append(key)

        res = []
        for i in range(len(arr) - 1, 0, -1):
            for num in arr[i]:
                res.append(num)
                if len(res) == k:
                    return res

        

        
        

            

        


        

File: Data Structures & Algorithms/split-array-largest-sum/submission-5.go
func splitArray(nums []int, k int) int {
    canSplit := func(largest int) bool {
        subarray := 1
        curSum := 0
        for _, num := range nums {
            curSum += num
            if curSum > largest {
                subarray++
                if subarray > k {
                    return false
                }
                curSum = num
            }
        }
        return true
    }

    l, r := 0, 0
    for _, num := range nums {
        if num > l {
            l = num
        }
        r += num
    }
    res := r

    for l <= r {
        mid := l + (r-l)/2
        if canSplit(mid) {
            res = mid
            r = mid - 1
        } else {
            l = mid + 1
        }
    }
    return res
}